vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -52,7 +52,18 @@ def scaling
           scale_parameters_formatted << {"name" => p.name, "value" => scale_parameters[p.name]}
         end
       end
-      if scale_parameters_formatted.length > 0
+
+      begin
+        # Check if stack is ready to be updated
+        update_ready = @stack.update_ready?(@stack.raw_status.first)
+      rescue => ex
+        log_and_flash_message(_("Unable to initiate scaling, obtaining of status failed: #{ex}"))
+        return
+      end
+
+      if !update_ready
+        add_flash(_("Provider is not ready to be scaled, another operation is in progress."), :error)
```

Comment:
Can I start refactoring task for this? That would be done next release, for this release, we would have just bugfix. I would rather not do the refactoring now, before the release, cause that could cause more bugs. I need to first grasp the idea, I didn't write this piece of code.